src/common/event-emitter.ts

```diff
@@ -0,0 +1,23 @@
+export type EventListener = (...args: any[]) => void
```

Comment:
Can you use `unknown[]` here?

Reply:
If we use `unknown[]` here, we end up with type errors here:

https://github.com/fission-suite/webnative/blob/6464460024c91e47b3cfba12681f77e7c24b2acc/src/auth/linking/consumer.ts#L119

For example, `Type 'unknown' is not assignable to type '{ pin: number[]; }'` which comes from the `OnChallenge` type.

I tried experimenting with generics like this:

```typescript
export type EventListener = <T>(...args: T[]) => void
```
```typescript
export type AccountLinkingRequestor = {
  on: OnChallenge<ChallengeArgs> & OnLink<LinkArgs> & OnDone
  cancel: () => void
}

type ChallengeArgs = {
  pin: number[]
}

type OnChallenge<ChallengeArgs> = (event: "challenge", listener: (args: ChallengeArgs) => void) => void
```
but that results in a similar type error: `Type 'T' is not assignable to type 'ChallengeArgs'`

Any other ideas on how to type this?

One thing to consider is the hints displayed in VSCode. For example, it would be better for them to see `{ pin: number[]; }` instead of `ChallengeArgs`.

Reply:
We converted this interface to use an event maps to type the event emitter in https://github.com/fission-suite/webnative/pull/335/commits/2470765a1387e917bf1845ee228c49c9b4e19a17.